### PR TITLE
[materialx] Add various features to control project options

### PR DESCRIPTION
**Pull request by theblackunknown**
If this PR updates an existing port, please uncomment and fill out this checklist:

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Cheney-W** [comment]
Please refrain from modifying newline characters in the portfile.cmake file, as this will result in additional unnecessary changes in the "file changed" output.

File: ports/materialx/portfile.cmake [hunk 1]
```diff
@@ -1,49 +1,65 @@
...
+    "${CURRENT_PACKAGES_DIR}/README.md"
+    "${CURRENT_PACKAGES_DIR}/THIRD-PARTY.md"
+    "${CURRENT_PACKAGES_DIR}/debug/CHANGELOG.md"
+    "${CURRENT_PACKAGES_DIR}/debug/LICENSE"
+    "${CURRENT_PACKAGES_DIR}/debug/README.md"
+    "${CURRENT_PACKAGES_DIR}/debug/THIRD-PARTY.md"
+    "${CURRENT_PACKAGES_DIR}/debug/include"
+
+    # MaterialXGenShader explicitely install resources
+    "${CURRENT_PACKAGES_DIR}/resources"
```
Comment (Cheney-W) [inline ports/materialx/portfile.cmake:57]:
```suggestion
    "${CURRENT_PACKAGES_DIR}/resources" 
    "${CURRENT_PACKAGES_DIR}/debug/resources"
```

**Comment by Cheney-W** [comment]
Pinging @theblackunknown Could you please address the review suggestions? Thanks!

**Comment by theblackunknown** [comment]
> Pinging @theblackunknown Could you please address the review suggestions? Thanks!

Hey @Cheney-W sorry for the delay, I'll address your suggestions this week !

> Please refrain from modifying newline characters in the portfile.cmake file, as this will result in additional unnecessary changes in the "file changed" output.

Do you mind if I keep those changes ? 
I find that having carriage return in files adds a lot of noise when reviewing with the GitHub UI.

**Comment by Cheney-W** [comment]
My intention is to keep the original newline characters CRLF used in the portfile.cmake, please do not replace them with LF.

**Comment by theblackunknown** [comment]
@Cheney-W I have made required changes, let me know if this is now good to go.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**